stdlib_test.go

```diff
@@ -48,214 +49,247 @@ func TestStdlibAdapterUsage(t *testing.T) {
+		},
+		{
+			regexp:     StdlibRegexpFull,
+			input:      `some prefix 2009/01/23 01:23:23.123123: hello`,
+			prefix:     "some prefix ",
+			joinPrefix: true,
+			want:       `ts="2009/01/23 01:23:23.123123" msg="some prefix hello"`,
+		},
+		{
+			regexp:     StdlibRegexpFull,
+			input:      `some prefix 2009/01/23 01:23:23.123123 /a/b/c/d.go:23: hello`,
+			prefix:     "some prefix ",
+			joinPrefix: true,
+			want:       `ts="2009/01/23 01:23:23.123123" caller=/a/b/c/d.go:23 msg="some prefix hello"`,
+		},
+		{
+			regexp:     StdlibRegexpFull,
+			input:      `some prefix 01:23:23.123123 /a/b/c/d.go:23: hello`,
+			prefix:     "some prefix ",
+			joinPrefix: true,
+			want:       `ts=01:23:23.123123 caller=/a/b/c/d.go:23 msg="some prefix hello"`,
+		},
+		{
+			regexp:     StdlibRegexpFull,
+			input:      `some prefix 2009/01/23 01:23:23 /a/b/c/d.go:23: hello`,
+			prefix:     "some prefix ",
+			joinPrefix: true,
+			want:       `ts="2009/01/23 01:23:23" caller=/a/b/c/d.go:23 msg="some prefix hello"`,
+		},
+		{
+			regexp:     StdlibRegexpFull,
+			input:      `some prefix 2009/01/23 /a/b/c/d.go:23: hello`,
+			prefix:     "some prefix ",
+			joinPrefix: true,
+			want:       `ts=2009/01/23 caller=/a/b/c/d.go:23 msg="some prefix hello"`,
+		},
+		{
+			regexp:     StdlibRegexpFull,
+			input:      `some prefix /a/b/c/d.go:23: hello`,
+			prefix:     "some prefix ",
+			joinPrefix: true,
+			want:       `caller=/a/b/c/d.go:23 msg="some prefix hello"`,
+		},
+		{
+			regexp:     StdlibRegexpFull,
+			input:      `/a/b/c/d.go:23: some prefix hello`,
+			prefix:     "some prefix ",
+			joinPrefix: true,
+			want:       `caller=/a/b/c/d.go:23 msg="some prefix hello"`,
+		},
+		{
+			regexp: StdlibRegexpDefault,
+			input:  `error encoding and sending metric family: write tcp 127.0.0.1:9182->127.0.0.1:60125: wsasend:`,
+			want:   `msg="error encoding and sending metric family: write tcp 127.0.0.1:9182->127.0.0.1:60125: wsasend:"`,
+		},
+		{
+			regexp: StdlibRegexpDefault,
+			input:  `2023/04/28 07:28:46 error encoding and sending metric family: write tcp 127.0.0.1:9182->127.0.0.1:60125: wsasend:`,
+			want:   `ts="2023/04/28 07:28:46" msg="error encoding and sending metric family: write tcp 127.0.0.1:9182->127.0.0.1:60125: wsasend:"`,
+		},
+		{
+			regexp: StdlibRegexpDefault,
+			input:  `2023/04/28 07:28:46 /a/b/c/d.go:23: error encoding and sending metric family: write tcp 127.0.0.1:9182->127.0.0.1:60125: wsasend:`,
+			want:   `ts="2023/04/28 07:28:46" msg="/a/b/c/d.go:23: error encoding and sending metric family: write tcp 127.0.0.1:9182->127.0.0.1:60125: wsasend:"`,
+		},
+		{
+			regexp: StdlibRegexpDefault,
+			input:  `2009/01/23 01:23:23.123123 /a/b/c/d.go:23: hello`,
+			want:   `ts="2009/01/23 01:23:23.123123" msg="/a/b/c/d.go:23: hello"`,
+		},
+		{
+			regexp: StdlibRegexpDefault,
+			input:  `1:9182f`,
+			want:   `msg=1:9182f`,
+		},
 	}
-}
+	for _, tt := range testcases {
+		t.Run(tt.input, func(t *testing.T) {
```

Comment:
I appreciate the use of `t.Run` here, but the `tt.input` values make for horrible test names.
- Many have embedded `/`'s which the `testing` package uses as the separator for test/sub-test/sub-sub-test/... names. The extra `/`'s in these test names confuses quite a lot of tooling that parses test names.
- They are not all unique so the testing package appends `#01` to the duplicates.
- Some of them are quite long and impractical to use with `go test -run=??`. Granted that's probably not a big need for these tests, but VS Code (and maybe other IDE's) now has this nifty test explorer that lets you rerun (or debug) individual sub-tests with the click of a button. That tool gets really confused by these test names, mostly due to the extra `/`'s, but on my Windows machine, a test name containing `.go:23:` for example seems to trigger a weird bug in the it's rendering of test ouput and I get things like this:

```
    --- PASS: TestStdLibAdapter/some_prefix_2009/01/23_01:23:23.123123:_hello#01 (0.00s)
    --- PASS: c:\Users\Chris\proj\go-kit\log\TestStdLibAdapter\some_prefix_2009\01\23_01:23:23.123123_\a\b\c\d.go:23:_hello#01 (0.00s)
```

This is clearly a bug in the VS Code tool, running `go test` directly doesn't reproduce that weird behavior, but we aren't doing it any favors with these test names.

For the record here are all the test names as they render to the console.

```
$ go test -v -run=TestStdLibAdapter
=== RUN   TestStdLibAdapter
=== RUN   TestStdLibAdapter/2009/01/23_01:23:23.123123_/a/b/c/d.go:23:_hello_world
=== RUN   TestStdLibAdapter/hello
=== RUN   TestStdLibAdapter/2009/01/23:_hello
=== RUN   TestStdLibAdapter/2009/01/23_01:23:23:_hello
=== RUN   TestStdLibAdapter/01:23:23:_hello
=== RUN   TestStdLibAdapter/2009/01/23_01:23:23.123123:_hello
=== RUN   TestStdLibAdapter/2009/01/23_01:23:23.123123_/a/b/c/d.go:23:_hello
=== RUN   TestStdLibAdapter/01:23:23.123123_/a/b/c/d.go:23:_hello
=== RUN   TestStdLibAdapter/2009/01/23_01:23:23_/a/b/c/d.go:23:_hello
=== RUN   TestStdLibAdapter/2009/01/23_/a/b/c/d.go:23:_hello
=== RUN   TestStdLibAdapter//a/b/c/d.go:23:_hello
=== RUN   TestStdLibAdapter/some_prefix_hello
=== RUN   TestStdLibAdapter/some_prefix_2009/01/23:_hello
=== RUN   TestStdLibAdapter/some_prefix_2009/01/23_01:23:23:_hello
=== RUN   TestStdLibAdapter/some_prefix_01:23:23:_hello
=== RUN   TestStdLibAdapter/some_prefix_2009/01/23_01:23:23.123123:_hello
=== RUN   TestStdLibAdapter/some_prefix_2009/01/23_01:23:23.123123_/a/b/c/d.go:23:_hello
=== RUN   TestStdLibAdapter/some_prefix_01:23:23.123123_/a/b/c/d.go:23:_hello
=== RUN   TestStdLibAdapter/some_prefix_2009/01/23_01:23:23_/a/b/c/d.go:23:_hello
=== RUN   TestStdLibAdapter/some_prefix_2009/01/23_/a/b/c/d.go:23:_hello
=== RUN   TestStdLibAdapter/some_prefix_/a/b/c/d.go:23:_hello
=== RUN   TestStdLibAdapter//a/b/c/d.go:23:_some_prefix_hello
=== RUN   TestStdLibAdapter/some_prefix_hello#01
=== RUN   TestStdLibAdapter/some_prefix_2009/01/23:_hello#01
=== RUN   TestStdLibAdapter/some_prefix_2009/01/23_01:23:23:_hello#01
=== RUN   TestStdLibAdapter/some_prefix_01:23:23:_hello#01
=== RUN   TestStdLibAdapter/some_prefix_2009/01/23_01:23:23.123123:_hello#01
=== RUN   TestStdLibAdapter/some_prefix_2009/01/23_01:23:23.123123_/a/b/c/d.go:23:_hello#01
=== RUN   TestStdLibAdapter/some_prefix_01:23:23.123123_/a/b/c/d.go:23:_hello#01
=== RUN   TestStdLibAdapter/some_prefix_2009/01/23_01:23:23_/a/b/c/d.go:23:_hello#01
=== RUN   TestStdLibAdapter/some_prefix_2009/01/23_/a/b/c/d.go:23:_hello#01
=== RUN   TestStdLibAdapter/some_prefix_/a/b/c/d.go:23:_hello#01
=== RUN   TestStdLibAdapter//a/b/c/d.go:23:_some_prefix_hello#01
=== RUN   TestStdLibAdapter/error_encoding_and_sending_metric_family:_write_tcp_127.0.0.1:9182->127.0.0.1:60125:_wsasend:
=== RUN   TestStdLibAdapter/2023/04/28_07:28:46_error_encoding_and_sending_metric_family:_write_tcp_127.0.0.1:9182->127.0.0.1:60125:_wsasend:
=== RUN   TestStdLibAdapter/2023/04/28_07:28:46_/a/b/c/d.go:23:_error_encoding_and_sending_metric_family:_write_tcp_127.0.0.1:9182->127.0.0.1:60125:_wsasend:
=== RUN   TestStdLibAdapter/2009/01/23_01:23:23.123123_/a/b/c/d.go:23:_hello#01
=== RUN   TestStdLibAdapter/1:9182f
```

Ideally we would write better names for each of these that describe semantically what they are testing, but I can see that won't be easy. The existing approach doesn't seem acceptable to me, though.

Suggestions?

Reply:
> and maybe other IDE's

It worked well with Goland on a mac. I didn't have any issues running single tests from IDE. Its more less talking about changing because of bugs in VS Code. 

I can't provide any well Suggestions here, because I'm not aware if they still cause problems on VSCode.

My Suggestions here: I will remove the `t.Run` here, because I have the personal feeling that I'm in feedback loop of 2 maintainer which never ends and I suggest that you going to fix this in a separate PR.

Reply:
An other Suggestions would that you proving the code that works.

The option "Allow edits by maintainers" is enabled, feel free to provide a solution.